BeerOverflowWindowsApp/BarProviders/GetBarListTripAdvisor.cs

```diff
@@ -22,116 +22,125 @@ class GetBarListTripAdvisor : IBeerable
-            List<BarData> barList = null;
-            try
+            var placeList = GetBarData(latitude, longitude);
+            var barList = PlaceListToBarList(placeList, radius);
+            return barList;
+        }
+
+        private IEnumerable<PlaceInfo> GetBarData(string latitude, string longitude)
+        {
+            var categories = _categoryListString.Split(',');
+            var placeList =  new List<PlaceInfo>();
+
+            foreach (var category in categories)
             {
-                var result = GetBarData(latitude, longitude);
-                barList = ApiQueryResponseToBars(result, radius);
+                var link = string.Format(_mapperLink, latitude, longitude, _accessKey, category);
+                var jsonStream = GetJsonStream(link);
+                placeList.AddRange(JsonConvert.DeserializeObject<PlacesResponse>(jsonStream).data);
             }
-            catch (Exception exception)
+            FetchLocations(placeList);
+
+            return placeList;
+        }
+
+        private void FetchLocations(IEnumerable<PlaceInfo> placeList)
+        {
+            foreach (var place in placeList)
             {
-                throw exception;
+                place.locationResponse = GetLocationForPlace(place);
             }
+        }
+
+        private LocationResponse GetLocationForPlace(PlaceInfo place)
+        {
+            var link = string.Format(_locationApiLink, place.location_id, _accessKey);
+            var jsonStream = GetJsonStream(link);
+            var placeLocation = JsonConvert.DeserializeObject<LocationResponse>(jsonStream);
+            return placeLocation;
+        }
+
+        public async Task<List<BarData>> GetBarsAroundAsync(string latitude, string longitude, string radius)
+        {
+            RegexTools.LocationDataTextIsCorrect(latitude, longitude, radius);
+            var placeList = await GetBarDataAsync(latitude, longitude);
+            var barList = PlaceListToBarList(placeList, radius);
             return barList;
         }
 
-        private PlacesResponse GetBarData(string latitude, string longitude)
+        private async Task<List<PlaceInfo>> GetBarDataAsync(string latitude, string longitude)
         {
-            PlacesResponse result = null;
-            try
-            {
-                var categories = _categoryListString.Split(',');
-                var webClient = new WebClient();
-                result = new PlacesResponse() { data = new List<PlaceInfo>() };
+            var categories = _categoryListString.Split(',');
+            var placeList = new List<PlaceInfo>();
 
-                foreach (var category in categories)
-                {
-                    var response = webClient.DownloadString(string.Format(_mapperLink, latitude, longitude, _accessKey, category));
-                    result.data.AddRange(JsonConvert.DeserializeObject<PlacesResponse>(response).data);
-                }
-                FetchLocations(result);
```

Comment:
As I mentioned, here's no empty line. No doubt we have to agree on some kind of standard